src/rpcmining.cpp

```diff
@@ -188,19 +188,13 @@ Value generate(const Array& params, bool fHelp)
                                               pblock->nNonce.size());
 
             // (x_1, x_2, ...) = A(I, V, n, k)
-            std::set<std::vector<unsigned int>> solns;
-            EhBasicSolveUncancellable(n, k, curr_state, solns);
-
-            for (auto soln : solns) {
-                bool isValid;
-                EhIsValidSolution(n, k, curr_state, soln, isValid);
-                assert(isValid);
+            std::function<bool(std::vector<eh_index>)> validBlock =
+                    [&pblock](std::vector<eh_index> soln) {
                 pblock->nSolution = soln;
-
-                if (CheckProofOfWork(pblock->GetHash(), pblock->nBits, Params().GetConsensus())) {
-                    goto endloop;
-                }
-            }
+                return CheckProofOfWork(pblock->GetHash(), pblock->nBits, Params().GetConsensus());
+            };
+            if (EhBasicSolveUncancellable(n, k, curr_state, validBlock));
```

Comment:
Can you spot the problem with this line? :mag_right:

This is a good example of an easily-overlooked change that can subtly change the execution logic. Or not-so-subtly in this case, thanks to the tests!